PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/com/microsoft/azuretools/container/ConsoleLogger.java

```diff
@@ -0,0 +1,95 @@
+ * the Software.
+ *
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+ * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.microsoft.azuretools.container;
+
+import org.eclipse.swt.graphics.Color;
+import org.eclipse.ui.IWorkbenchPage;
+import org.eclipse.ui.IWorkbenchWindow;
+import org.eclipse.ui.PartInitException;
+import org.eclipse.ui.PlatformUI;
+import org.eclipse.ui.console.ConsolePlugin;
+import org.eclipse.ui.console.IConsole;
+import org.eclipse.ui.console.IConsoleConstants;
+import org.eclipse.ui.console.IConsoleManager;
+import org.eclipse.ui.console.IConsoleView;
+import org.eclipse.ui.console.MessageConsole;
+import org.eclipse.ui.console.MessageConsoleStream;
+
+public class ConsoleLogger {
+    private MessageConsole console = null;
+    public MessageConsoleStream out = null;
+    public MessageConsoleStream err = null;
+
+    private ConsoleLogger() {
+        // TODO: make console name constant in future
+        console = findConsole(Constant.CONSOLE_NAME);
+        out = console.newMessageStream();
+        err = console.newMessageStream();
+        out.setColor(new Color(null, 0, 0, 255));
+        err.setColor(new Color(null, 255, 0, 0));
+
+        IConsole myConsole = this.console;
+        IWorkbenchWindow window = PlatformUI.getWorkbench().getActiveWorkbenchWindow();
+        IWorkbenchPage page = window.getActivePage();
+        String id = IConsoleConstants.ID_CONSOLE_VIEW;
+        IConsoleView view;
+        try {
+            view = (IConsoleView) page.showView(id);
+            view.display(myConsole);
+        } catch (PartInitException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private static class LazyHolder {
+        static final ConsoleLogger INSTANCE = new ConsoleLogger();
+    }
+
+    public static ConsoleLogger getInstance() {
+        return LazyHolder.INSTANCE;
+    }
+
+    public static void info(String infoMsg) {
+        LazyHolder.INSTANCE.out.println(String.format("[INFO]\t%s", infoMsg));
+    }
+
+    public static void error(String errorMsg) {
+        LazyHolder.INSTANCE.err.println(String.format("[ERROR]\t%s", errorMsg));
+    }
+
+    private MessageConsole findConsole(String name) {
+        ConsolePlugin plugin = ConsolePlugin.getDefault();
+        IConsoleManager conMan = plugin.getConsoleManager();
+        IConsole[] existing = conMan.getConsoles();
+        for (int i = 0; i < existing.length; i++) {
+            System.out.println(existing[i].getType());
+            if (name.equals(existing[i].getName()) && existing[i].getType().equals(Constant.CONSOLE_TYPE)) {
```

Comment:
Can we use existing[i] instance of MessageConsole? instead of store the console_type. The problem here is you use the stored console_type, but cast to "MessageConsole"